editor/filesystem_dock.cpp

```diff
@@ -1923,24 +1943,21 @@ void FileSystemDock::_file_option(int p_option, const Vector<String> &p_selected
 		} break;
 
 		case FILE_RENAME: {
-			// Rename the active file.
-			if (!p_selected.is_empty()) {
+			if (tree->is_anything_selected() && !p_selected.is_empty()) {
+				// Set to_rename variable for callback execution.
 				to_rename.path = p_selected[0];
-				if (to_rename.path != "res://") {
-					to_rename.is_file = !to_rename.path.ends_with("/");
-					if (to_rename.is_file) {
-						String name = to_rename.path.get_file();
-						rename_dialog->set_title(TTR("Renaming file:") + " " + name);
-						rename_dialog_text->set_text(name);
-						rename_dialog_text->select(0, name.rfind("."));
-					} else {
-						String name = to_rename.path.substr(0, to_rename.path.length() - 1).get_file();
-						rename_dialog->set_title(TTR("Renaming folder:") + " " + name);
-						rename_dialog_text->set_text(name);
-						rename_dialog_text->select(0, name.length());
-					}
-					rename_dialog->popup_centered(Size2(250, 80) * EDSCALE);
-					rename_dialog_text->grab_focus();
+				to_rename.is_file = !to_rename.path.ends_with("/");
+
+				// Edit node in Tree.
+				tree->grab_focus();
+				tree->edit_selected();
+
+				if (to_rename.is_file) {
+					String name = to_rename.path.get_file();
+					tree->set_editor_selection(0, name.rfind("."));
+				} else {
+					String name = to_rename.path.substr(0, to_rename.path.length() - 1).get_file();
```

Comment:
This can be replaced with either
```suggestion
					String name = to_rename.path.left(-1).get_file();
```
(something like that)
or
```suggestion
					String name = to_rename.path.trim_suffix("/").get_file();
```